app/src/main/java/com/asfoundation/wallet/my_wallets/change_wallet/ChangeActiveWalletDialogFragment.kt

```diff
@@ -52,12 +53,32 @@ class ChangeActiveWalletDialogFragment : DaggerBottomSheetDialogFragment(),
   override fun getTheme(): Int = R.style.AppBottomSheetDialogTheme
 
   override fun onStateChanged(state: ChangeActiveWalletState) {
-    setWalletDetails(state.walletBalance)
+    when (state.changeWalletAsync) {
+      Async.Uninitialized -> setWalletDetails(state.walletBalance)
+      is Async.Loading -> setLoading()
+      is Async.Fail -> Unit
+      is Async.Success -> Unit
```

Comment:
shouldn't Success be used to hide the bottomSheet after the change?

Reply:
This is already done on ViewModel through a side effect